File: Data Structures & Algorithms/longest-consecutive-sequence/submission-9.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> seen=new HashSet<>();
        Set<Integer> set = new HashSet<>();
        int maxcnt=0;
        for (int num:nums){
            set.add(num);
        }
        for(int i=0;i<nums.length;i++){
            if (seen.contains(nums[i])){
                continue;
            }
            int cnt=0;
            int curr=nums[i];
            while (set.contains(curr)){
                cnt+=1;
                curr+=1;
                seen.add(curr);
                maxcnt=Math.max(maxcnt,cnt);                
            }
        }
        return maxcnt;


    }
}

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-9.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        if len(nums)==1:
            if nums[0]==target:
                return 0
            else:
                return -1
        l=0
        r=len(nums)-1

        while r>=l:            
            m=(r+l)//2
            mid=nums[m]
            if mid==target:
                return m
            if nums[l]<=mid:
                if nums[l]<=target<mid:
                    r=m-1
                else:
                    l=m+1
            else:
                if mid<target<=nums[r]:
                    l=m+1
                else:
                    r=m-1

        return -1
